apps/go-mod-validator/test/deps.test.ts

```diff
@@ -0,0 +1,75 @@
+import { execSync } from "child_process";
+import { getDependenciesMap } from "../src/deps";
+import { beforeEach, describe, expect, it, vi, Mock } from "vitest";
+
+vi.mock("child_process");
+
+describe("getDependenciesMap", () => {
+  it("Successful: should return a map of <go.mod files: dependencies in json>", () => {
+    const paths = ["/path/to/first/go.mod", "/path/to/second/go.mod"];
+    const goListMockOutput1 =
+      '[{"Path": "github.com/smartcontractkit/go-plugin", "Version": "v0.0.0-20240208201424-b3b91517de16"}]';
+    const goListMockOutput2 =
+      '[{"Path": "github.com/smartcontractkit/grpc-proxy", "Version": "v0.0.0-20230731113816-f1be6620749f"}]';
+
+    let callCount = 0;
+    (execSync as Mock).mockImplementation((command) => {
+      if (command.includes("go list")) {
+        callCount++;
+        return callCount === 1 ? goListMockOutput1 : goListMockOutput2;
+      } else if (command.includes("find")) {
+        return paths.join("\n");
+      }
+      return "";
+    });
+
+    const result = getDependenciesMap("");
+    expect(result).toEqual(
+      new Map<string, any>([
+        [paths[0], JSON.parse(goListMockOutput1)],
+        [paths[1], JSON.parse(goListMockOutput2)],
+      ]),
+    );
+  });
+
+  it("Fail: should handle no go.mod files found", () => {
```

Comment:
```suggestion
  it("should handle no go.mod files found", () => {
```
nit: we dont list test cases by their expected boolean result, knowing that it handles errors implies that the function fails

Reply:
Done